extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js

```diff
@@ -482,13 +489,42 @@ export class AmpLightboxViewer extends AMP.BaseElement {
   }
 
   /**
-   * Set up container listener.
+   * Set up event listeners.
    * @private
    */
-  setupContainerListener_() {
+  setupEventListeners_() {
     dev().assert(this.container_);
     const toggleControls = this.toggleControls_.bind(this);
-    listen(dev().assertElement(this.container_), 'click', toggleControls);
+    this.unlistenClick_ = listen(dev().assertElement(this.container_),
+        'click', toggleControls);
+
+    // iOS non-safari browsers do not reliably fire onResize on orientation
+    // change, so listen to orientationchange to trigger resize
+    const platform = Services.platformFor(this.win);
+    if (platform.isIos() && !platform.isSafari()) {
+      // The debounce is to force a 500ms delay due to Webkit bug #170595
+      const debouncedOnResize = debounce(this.win,
+          this.onResize_.bind(this), 500);
+      this.unlistenOrientationChange_ = listen(this.win,
```

Comment:
Could this be closer to the event registration of the `resize` event?

It is confusing that the `onResize` call is somewhere else.

Reply:
I tried to address this by moving this block that adds a `orientationchange` listener to the `registerOnResizeHandler_()` function. An alternative approach here, since the `resizeCurrentImageViewer` function is a class function, is actually to register the resize handler once and for all (for the entire duration for which the lightbox is open), instead of registering and deregistering it on every slide change, and having the `resizeCurrentImageViewer` be more robust to error conditions. I think that approach makes prettier code. Sadly iOS Chrome and FF takes very long to test... and there were some weird mystery bugs with that approach the last time I tried. I'm open to trying to make that work, but I also think a lot of this code will eventually change some more after we review the API and review the component for performance.